### ISSUE TITLE
Market order precision and funds parameter

### ISSUE THREAD

**Issue by LorenzE** [coinbase/gdax-tt#62]
Hi there,

I recently started to use your toolbox to code my first trading bot on GDAX. This is my first time coding with typescript. I quite enjoy it though and your toolbox is very straighforward to learn :) 

My problem is, that as soon as I want to give a more precise market order with more than two decimal places (e.g. 0.024) it will only send with a precision of two decimal places (e.g. 0.02 - the last decimal 4 is lost). I know that the minimum for a limit order is set at 0.01 but I thought there is no restriction with market orders. My function to perform the actual trade looks like this:

```javascript
function performTrade(side:string, price:BigJS) {   
    let size: BigJS = workingProductAmount;
    
    if(side == 'sell') {
        if(workWithWins) {
            size = balanceProduct.available; 
        }
    
        logger.log('info', `Time to sell ${size}${base}s at current price!`); 
    } else if (side == 'buy') {        
        if(workWithWins) {
            size = balanceEUR.available.div(price);
        }

        logger.log('info', `Time to buy ${size}${base}s at current price!`);   
    }
  
    logger.log('info', `size.toString() ${size.toString()}`);   

    const message: PlaceOrderMessage = {
        type: 'placeOrder',
        productId: product,
        size: size.toString(),
        side: side,
        orderType: 'market',
        time: null
    };   

    trader.executeMessage(message);
}
```

Also, I have trouble using the funds parameter in PlaceOrderMessage. The market order only works when I specify the size field and no value for funds, but it does not work vice versa. This is why I have to calculate the max amount of buyable BTC based on my current available EUR balance.

I am sure I am doing something wrong here.I would greatly appreciate if someone could point me to the right direction. 

Cheers,

Lorenz

**Comment by zhm** [comment]
Are you using `fitOrders: true` on the `Trader`? It looks like it truncates to 2 places by default: https://github.com/coinbase/gdax-tt/blob/master/src/core/Trader.ts#L66-L67

So you can try setting your own `sizePrecision` and `pricePrecision` or disable `fitOrders`.

**Comment by CjS77** [comment]
Hey @LorenzE 

Welcome! One of our missions is to make the GTT easy to learn and use, so it seems we're on the right track.. somewhat :)

@zhm is right, you must specify `sizePrecision` to a higher value, or set `fitOrder` to false in your TraderConfig.

BTW, you're also able to use a `GDAXExchangeAPI` instance to place orders etc directly. The idea behind `Traders`  is that it hooks up into a stream setup to place trades based on incoming messages (and maybe this is exactly what you want to do, in which case carry on).

I'm working on a fairly large update (0.2.0) that will include a MarketMaker bot which makes extensive use of this idea.

**Comment by CjS77** [comment]
WRT your funds question, I'm guessing you've surfaced a bug in Trader wrt Market orders
which has been fixed in #66 and v0.1.20.

**Comment by LorenzE** [comment]
Thx @zhm and @CjS77. I got it working now :)
